components/user-mgt/org.wso2.carbon.user.mgt.ui/src/main/java/org/wso2/carbon/user/mgt/ui/client/ClaimDataAdminClient.java

```diff
@@ -53,12 +55,44 @@ public ClaimDataAdminClient(String cookie, String backendServerURL, Configuratio
         option.setProperty(org.apache.axis2.transport.http.HTTPConstants.COOKIE_STRING, cookie);
     }
 
-    public LocalClaimDTO[] getLocalClaims() throws RemoteException, ClaimMetadataManagementServiceClaimMetadataException {
+    public LocalClaimDTO[] getLocalClaims()
+            throws RemoteException, ClaimMetadataManagementServiceClaimMetadataException {
 
         LocalClaimDTO[] localClaimDTOs = stub.getLocalClaims();
         if (localClaimDTOs == null) {
             localClaimDTOs = new LocalClaimDTO[0];
         }
         return localClaimDTOs;
     }
+
+    public String getRegex(LocalClaimDTO[] localClaimDTO, String claimURI) {
+
+        String pattern = "";
+        for (LocalClaimDTO localClaim : localClaimDTO) {
+            if (claimURI.equals(localClaim.getLocalClaimURI())) {
+                ClaimPropertyDTO[] claimPropertyDTOs = localClaim.getClaimProperties();
+                if (claimPropertyDTOs != null) {
+                    for (ClaimPropertyDTO claimPropertyDTO : claimPropertyDTOs) {
+                        if (ClaimConstants.REGULAR_EXPRESSION_PROPERTY.equals(claimPropertyDTO.getPropertyName())) {
+                            pattern = claimPropertyDTO.getPropertyValue();
+                            pattern = "/" + pattern + "/";
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+        return pattern;
+    }
+
+    public String getRegex(String claimURI) throws RemoteException {
+
+        LocalClaimDTO[] localClaims = new LocalClaimDTO[0];
+        try {
+            localClaims = getLocalClaims();
+        } catch (ClaimMetadataManagementServiceClaimMetadataException e) {
+            throw new RemoteException("Error while accessing claims");
```

Comment:
I don't see throwing a remote exception is good. If you do not have a dedicated exception to be thrown from clients you can define one